### ISSUE TITLE
Export list of forsyth positions

### ISSUE THREAD

**Issue by andrew-buchanan** [hwatheod/retractor#6]
Hi thanks for all your work.

It would be very valuable if, following a solve, one can export a text list in forsyth format of all the candidate positions. A user can then easily braid these into a Popeye input file, and properly C+ help retractors, which cannot at the moment be solved automatically. 

There are nearly 2,000 such problems at the moment in PDB. Some are marked solved by an antique program called RSP from abut 25 years ago, but that does not seem to be currently available or supported.

So this simple export would be a huge win. Thanks so much!

**Comment by hwatheod** [comment]
I am not entirely clear on what is being asked here. Is it the entire tree of positions that Retractor searched? Or just the first level?  Or something else entirely?

**Comment by andrew-buchanan** [comment]
Hi thanks for reply. 

Here's an example:
https://pdb.dieschwalbe.de/P1183936.
If I put the position into Retractor, I get 100 possible last moves.
I would like an export file which comprises 100 lines, each with a Forsythe diagram, e.g. one line retracting Ke2xQd2 would be:
3k4/8/8/1N3P2/3B4/8/3qK3/8

I can then take these 100 lines and splice them into a Popeye input script, and prove the soundness of this help retractor. 
There are over 1700 help retractors, including 150 by the famous lady composer, Edith Baird. They could all be solved by chaining Retractor to Popeye in this way.

One issue: Retractor's output is currently truncated at 100, and there's no way to "cycle through" to see the ones missed in the first pass. Search will be shallow for for most help retractors - just one or two single moves, and false positives don't matter because even if one turns out to generate a forward solution, it will be checked by hand at this point. Indeed such a discovery would be a retro try - an additional feature of the problem. But we do need *all* of the possible 1 move retractions, please, without truncation.

Can your internal throttle be based on a different metric than the number of parallel single move retractions please?

Bottom line: implementing this feature request requires no more "intelligence" on the part of Retractor, and yet would render a large class of retro problems solvable. Please implement this.

**Comment by andrew-buchanan** [comment]
Hi - is more clarity required, please?

In a nutshell:

Can export text file with n lines, each comprising one FEN diagram, which represent all the positions possible by retracting 1 (or 2) single moves?

Most of Baird's problems seem to retract 2 single moves, I think that would be enough. 3+ retractions is very uncommon in help retractors.

Thanks,
Andrew

**Comment by hwatheod** [comment]
Yes, I understand what you are asking. I haven't had time to work on it.

**Comment by hwatheod** [comment]
Feature is available in version 2.2.1.
